test/config-description.js

```diff
@@ -52,18 +68,55 @@ describe('config: description', function() {
   it('Should configure with a .gulp.* file in cwd by --cwd', function(done) {
     runner({ verbose: false })
       .basedir(fixturesDir)
-      .chdir('qux')
-      .gulp('--tasks', '--gulpfile ../foo/bar/gulpfile.js', '--cwd .')
+      .chdir('findup/currentdir')
+      .gulp('--tasks', '--cwd ../../cwd')
       .run(cb);
 
     function cb(err, stdout, stderr) {
       expect(err).toEqual(null);
       expect(stderr).toEqual('');
-      var expected = fs.readFileSync(path.join(expectedDir, 'output1.txt'),
-        'utf-8');
-      stdout = eraseTime(stdout);
+      var expected = fs.readFileSync(path.join(expectedDir, 'output0.txt'), 'utf-8');
+      stdout = skipLines(eraseTime(stdout), 1);
       expect(stdout).toEqual(expected);
       done(err);
     }
   });
+
+  it('Should configure with a .gulp.* file having renamed gulpfile', function(done) {
+    runner({ verbose: false })
+      .basedir(fixturesDir)
+      .chdir('initcwd/renamedgulpfile')
+      .gulp('--tasks')
+      .run(cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
+      expect(stderr).toEqual('');
+      var expected = fs.readFileSync(path.join(expectedDir, 'output3.txt'), 'utf-8');
+      stdout = skipLines(eraseTime(stdout), 1);
+      expect(stdout).toEqual(expected);
+      done(err);
+    }
+  });
+
+  it('Should give priority to .gulp.* in cwd higher than .gulp.* in initial cwd', function(done) {
+    runner({ verbose: false })
+      .basedir(fixturesDir)
+      .chdir('initcwd/renamedgulpfile')
+      .gulp('--tasks --cwd ../../cwd')
+      .run(cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toEqual(null);
+      expect(stderr).toEqual('');
+      var expected0 = fs.readFileSync(path.join(expectedDir, 'output0.txt'), 'utf-8');
+      var expected3 = fs.readFileSync(path.join(expectedDir, 'output3.txt'), 'utf-8');
+      stdout = skipLines(eraseTime(stdout), 1);
+
+      expect(headLines(stdout, 1)).toEqual(headLines(expected0, 1));
+      expect(skipLines(stdout, 1)).toEqual(skipLines(expected3, 1));
```

Comment:
The behavior of this seems extremely awkward.  When using `--cwd`, I wouldn't expect the `flags.gulpfile` to be picking up a gulpfile where I ran my command - I'd expect it to run a gulpfile where my `--cwd` was pointing.